### PR TITLE
feat: Rename AI Assistant to n8n Assistant

### PR DESCRIPTION
**Pull request by Tuukkaa**
## Summary

Rename the Instance AI product from "AI Assistant" to "n8n Assistant" in every surface a user reads.

- **UI copy.** 68 keys in `en.json`. The legacy assistant keys (`aiAssistant.*`, `settings.ai.*`) are unchanged.
- **Hardcoded frontend strings.** The module descriptor name, the sandbox credential name in the settings view, and the agent session timeline label. The timeline view now reads the label from the existing `agentSessions.origin.instanceAi` key instead of a literal.
- **Backend messages.** The "no model configured" error, the out-of-credits message, and the authorization log lines.
- **Credential names.** The settings service now creates "n8n Assistant model", "n8n Assistant web search", and "n8n Assistant sandbox" for new connections. Connections that exist today keep their current name: the settings service keeps a credential's name when it updates a connection, and the Instance AI team preferred not to rename existing rows with a migration.
- **Agent wording.** Skills, tools, guidance, and the system prompt's setup-card line now say "n8n Assistant panel" and "n8n Assistant help thread". The system prompt's opening self-description is unchanged, so the recorded Playwright expectations keep matching.
- **Telemetry.** Event descriptions are updated. Event names are unchanged, so dashboards and funnels keep working.
- **Docs, comments, Computer Use CLI help text, and Playwright locators.**

## How to test

Run the unit tests that assert the renamed strings:

```
cd packages/cli
pnpm vitest run src/modules/instance-ai src/credentials/__tests__/credentials.service.test.ts
cd ../@n8n/instance-ai
pnpm vitest run src/skills src/workflow-loop
```

Manual check: open the assistant panel, the onboarding wizard, and Settings > n8n Assistant. The name reads "n8n Assistant" everywhere. Add a new model connection and confirm its credential is named "n8n Assistant model". A connection that existed before the upgrade keeps its old name.

## Review / Merge checklist

- [x] I have seen this code, I have run this code, and I take responsibility for this code.
- [x] PR title and summary are descriptive.
- [ ] Docs updated or follow-up ticket created. The n8n docs still call the feature "AI Assistant".
- [x] Tests included. Unit tests assert the renamed strings and credential names.
- [ ] PR Labeled with `Backport to Beta`, `Backport to Stable`, or `Backport to v1` (if the PR is an urgent fix that needs to be backported)

🤖 Generated with [Claude Code](https://claude.com/claude-code)

https://claude.ai/code/session_019Erj6amuGnCReu4iz8M6hN

### CODE REVIEW

File: packages/frontend/@n8n/i18n/src/locales/en.json [hunk 1]
```diff
@@ -5267,7 +5267,7 @@
 	"settings.n8nConnect.wallet.quota": "budget",
 	"settings.n8nConnect.wallet.topUp": "Top up balance",
 	"settings.n8nConnect.usage.refresh.tooltip": "Refresh usage records",
-	"instanceAi.onboarding.title": "Build and debug faster with the AI Assistant",
+	"instanceAi.onboarding.title": "Build and debug faster with the n8n Assistant",
```
[the remark this answers is not in the document]
Comment (Tuukkaa) [reply packages/frontend/@n8n/i18n/src/locales/en.json:5224]: These four strings are intentionally unchanged. They belong to the legacy assistant, not to Instance AI: `aiAssistant.builder.creditBanner.limitReachedText` and the three `settings.ai.*` data-sharing strings describe the older cloud assistant and AI Builder. The rename is scoped to the Instance AI product; the legacy feature keeps its name.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: packages/@n8n/db/src/migrations/common/1788863190206-RenameInstanceAiCredentials.ts [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+export class RenameInstanceAiCredentials1788863190206 implements ReversibleMigration {
+	async up(context: MigrationContext) {
+		for (const { from, to, credentialUseIds } of RENAMES) {
+			await this.rename(context, from, to, credentialUseIds);
+		}
+	}
+
+	async down(context: MigrationContext) {
+		for (const { from, to, credentialUseIds } of RENAMES) {
+			await this.rename(context, to, from, credentialUseIds);
```
[the remark this answers is not in the document]
Comment (Tuukkaa) [reply packages/@n8n/db/src/migrations/common/1788863190206-RenameInstanceAiCredentials.ts]: Fixed in 7bd84956 without recording ids. `down` now renames every instance credential that carries a new name, independent of the assignments. Only this version writes those names, so that set is exactly the rows `up` renamed plus any connection the new version created, and both need the old name after a rollback. The migration test covers a connection that was replaced between `up` and `down`.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: packages/@n8n/db/src/migrations/common/1788863190206-RenameInstanceAiCredentials.ts [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+	private async renameInstanceScoped(
+		{ escape, runQuery }: MigrationContext,
+		from: string,
+		to: string,
+	) {
+		const credentials = escape.tableName(CREDENTIALS_TABLE);
+		const name = escape.columnName('name');
+		const usageScope = escape.columnName('usageScope');
+
+		await runQuery(
```
Comment (RicardoE105) [inline packages/@n8n/db/src/migrations/common/1788863190206-RenameInstanceAiCredentials.ts]: Why we do not have the assignment check when running the down. This will mean that if we revert the migration some credential that did not need to be renamed back will be.

**Review by RicardoE105 — COMMENTED** [review]
left one comment. I will also assign someone from the instance AI team to make sure this is safe as they have much better context than me here.

File: packages/@n8n/db/src/migrations/common/1788863190206-RenameInstanceAiCredentials.ts (hunk 3 above)
Comment (Tuukkaa) [reply packages/@n8n/db/src/migrations/common/1788863190206-RenameInstanceAiCredentials.ts]: Good catch. The assignment check was left out of `down` on purpose, because the assignment can move between `up` and `down`: if an admin replaces the connection after the upgrade, the renamed row loses its assignment and an assignment-scoped `down` would miss it. Renaming by name alone fixed that but had the problem you describe.

e194d56f resolves both. `up` now records the ids of the rows it renamed in the `settings` table under `instanceAi.renamedCredentialIds`, and `down` restores exactly those rows and deletes the record. A credential that carries the new name but was never renamed keeps it, whether it is assigned or not, and a renamed credential is restored even after its assignment moved. The migration test covers both cases.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: packages/@n8n/db/src/migrations/common/1788863190206-RenameInstanceAiCredentials.ts [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+			`UPDATE ${credentials} SET ${nameColumn} = '${to}' WHERE ${nameColumn} = '${from}' AND ${idColumn} IN (${idList})`,
+		);
+	}
+
+	private async writeRenamedIds(context: MigrationContext, ids: string[]) {
+		const { escape, runQuery } = context;
+		await this.deleteRenamedIds(context);
+		await runQuery(
+			`INSERT INTO ${escape.tableName(SETTINGS_TABLE)} (${escape.columnName('key')}, ${escape.columnName('value')}, ${escape.columnName('loadOnStartup')}) ` +
```
Comment (RicardoE105) [inline packages/@n8n/db/src/migrations/common/1788863190206-RenameInstanceAiCredentials.ts]: We do not need to add anything to the settings, we can just add the missing filter (assigments) to the query we have in down. The last commit it's not the right approach

Comment (Tuukkaa) [reply packages/@n8n/db/src/migrations/common/1788863190206-RenameInstanceAiCredentials.ts]: Done in a9fa8cce. The settings record is gone, and `down` now runs the same query as `up` with the names swapped, so it only renames back the credentials that an Instance AI credential use is assigned to. The test covers an instance credential and a project credential that carry the new name by hand and stay untouched on rollback.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: packages/@n8n/db/src/migrations/common/1788863190206-RenameInstanceAiCredentials.ts [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+		const assignments = escape.tableName(ASSIGNMENTS_TABLE);
+		const name = escape.columnName('name');
+		const id = escape.columnName('id');
+		const credentialId = escape.columnName('credentialId');
+		const credentialUseId = escape.columnName('credentialUseId');
+		const useIdList = credentialUseIds.map((useId) => `'${useId}'`).join(', ');
+
+		await runQuery(
+			`UPDATE ${credentials} SET ${name} = '${to}' WHERE ${name} = '${from}' AND ${id} IN ` +
+				`(SELECT ${credentialId} FROM ${assignments} WHERE ${credentialUseId} IN (${useIdList}))`,
```
[the remark this answers is not in the document]
Comment (Tuukkaa) [reply packages/@n8n/db/src/migrations/common/1788863190206-RenameInstanceAiCredentials.ts]: This is a deliberate trade-off, decided by the maintainer. A recorded-ids variant (storing the renamed ids in the settings table) was tried in e194d56f and rejected by @RicardoE105 in favor of the current shape: `down` applies the same assignment filter as `up`, and nothing is written to settings. The rollback edge case is documented in the migration's doc comment, and the test `restores only the assigned credentials on rollback` covers the unassigned and project-scoped cases that stay untouched.

---
_Generated by [Claude Code](https://claude.ai/code)_

**Comment by Tuukkaa** [comment]
CI on a9fa8cce: `E2E / Shard 2` failed, which turns `Required Checks` red. The only failed test is `tests/e2e/ai/langchain-agents.spec.ts:169` ("should add multiple tool nodes to Agent node tool input type", multi-main project). The click on the "Wikipedia" node-creator item resolved to a visible and enabled element and then timed out after 20s, on all three attempts.

This failure is not from this PR:

- The PR does not touch the node creator, `CanvasPage.ts`, that spec, or any node in `nodes-base`. The Playwright changes are limited to the Instance AI page object, two Instance AI specs, one auth spec, and a comment.
- The E2E suite was green on the earlier head 7bd84956. The two commits since then change only the migration file and its migration test, which the frontend never loads.
- I found no open or merged PR that fixes this test, so there is no fix to port.

I am re-running the failed job once. If it fails again in the same way, I will report back here instead of widening the PR into the canvas page object.

---
_Generated by [Claude Code](https://claude.ai/code)_

**Comment by Tuukkaa** [comment]
CI on c6436f4e8f: `E2E / Shard 4` failed. The only failed test is `tests/e2e/settings/log-streaming/log-streaming-delivery.spec.ts:128` ("should deliver configured method and custom headers set via the UI", multi-main project). `page.goto('/settings/log-streaming')` did not reach the load event within 10s, on all three attempts and against two different stack ports. A sibling test in the same file was flaky in the same way and passed on retry.

This failure is not from this PR: the diff touches nothing in log streaming, the event bus, or that spec. All 15 other shards, the DB tests, and the unit tests passed on this head. I found no open or merged PR that fixes this spec, so there is no fix to port. I am re-running the failed job once.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: .agents/skills/community-pr-readiness-check/reference/teams.md [hunk 6]
```diff
@@ -54,7 +54,7 @@ Use the canonical owners script at `.github/scripts/owners.mjs`. It parses `.git
 | `design`                  | Design                  | `team:design`      |
 | `iam`                     | Identity & Access       | `team:identity`    |
 | `ligo`                    | Lifecycle & Governance  | `team:lifecycle`   |
-| `instance-ai`             | AI Assistant            | `team:instance-ai` |
+| `instance-ai`             | n8n Assistant           | `team:instance-ai` |
```
Comment (mutdmour) [inline .agents/skills/community-pr-readiness-check/reference/teams.md]: I think our team on github was renamed to ai-assistant

File: packages/@n8n/db/src/migrations/common/1788863190206-RenameInstanceAiCredentials.ts
Comment (mutdmour) [inline packages/@n8n/db/src/migrations/common/1788863190206-RenameInstanceAiCredentials.ts:1]: Do we really need this migration? Can we remove it and simply let the old names as they are? I don't think this is important enough, would rather avoid any sideeffects. For example workflows tend to cache the credential name.

Comment (Tuukkaa) [reply packages/@n8n/db/src/migrations/common/1788863190206-RenameInstanceAiCredentials.ts:1]: Fair point. The facts for the decision:

- Without the migration, connections that exist today keep the old credential name. The settings service keeps `current.name` when it updates a connection, so only a connection that an admin removes and adds again gets the new name.
- Instance AI credentials are instance-scoped and no workflow node references them, so the cached-name side effect should not apply here. The benefit of the migration is cosmetic either way: the credential names on the Settings page.
- Dropping it also removes the only `required` owner on this PR (migrations-review), so the PR needs one approval fewer.

I lean towards dropping it. Leaving this thread open for the author's call.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: .agents/skills/community-pr-readiness-check/reference/teams.md (hunk 6 above)
Comment (Tuukkaa) [reply .agents/skills/community-pr-readiness-check/reference/teams.md]: Fixed in 94028cc4. The slug is now `ai-assistant`, as in OWNERS. I also restored the Linear team name in that row to "AI Assistant": the Linear team kept its name, only the product was renamed.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: packages/@n8n/db/src/migrations/common/1788863190206-RenameInstanceAiCredentials.ts
Comment (mutdmour) [reply packages/@n8n/db/src/migrations/common/1788863190206-RenameInstanceAiCredentials.ts:1]: Okay then please drop it.

Comment (Tuukkaa) [reply packages/@n8n/db/src/migrations/common/1788863190206-RenameInstanceAiCredentials.ts:1]: Dropped in 490978d7. The migration and its test are removed. New connections get the "n8n Assistant" names; existing connections keep their current name. The PR description is updated to match.

---
_Generated by [Claude Code](https://claude.ai/code)_

**Review by mutdmour — DISMISSED** [review]

**Comment by Tuukkaa** [comment]
Merged `master` into this branch (6806d11b79) to resolve conflicts with #38031 in `post-build-flow/SKILL.md` and `runtime-skills.test.ts`. The resolution keeps the new setup panel text from master and applies the "n8n Assistant panel" rename inside it. No other changes.

The push dismissed the earlier approval, so this needs a re-approval once CI is green. Auto-merge (squash) is still enabled.

Context: the PR was approved and green since yesterday 15:29 UTC but the merge queue never picked it up, and the merge box showed "Required Checks — Expected" although that check had reported success on the head commit. The fresh head gives GitHub a new set of checks to evaluate.

---
_Generated by [Claude Code](https://claude.ai/code)_

**Review by mutdmour — APPROVED** [review]

**Comment by Tuukkaa** [comment]
**E2E / Shard 7 failed on 6806d11b79** in `tests/e2e/ai/assistant-support-chat.spec.ts` › "should not send workflow context if nothing changed" (project `multi-main:e2e`, 3 of 3 attempts). The second chat request carried `currentWorkflow` although nothing changed.

This is not a failure of this PR:
- The spec covers the legacy assistant. This PR does not change `features/ai/assistant` or that spec.
- The same spec ran under the same project in yesterday's green run on this branch (490978d7d2) and passed.
- Master's nightly full E2E run (run 34428183462, 02:07 UTC today) passed it. No master commit since then touches the legacy assistant code.
- The spec has a recorded flaky rate in `.github/test-metrics/playwright.json`.

No fix exists on master for it. Re-running the failed job once.

---
_Generated by [Claude Code](https://claude.ai/code)_
